pynest/nest/server/hl_api_server.py

```diff
@@ -60,11 +72,117 @@
 ]
 
 app = Flask(__name__)
-CORS(app)
+# Inform client-side user agents that they should not attempt to call our server from any
+# non-whitelisted domain.
+CORS(app, origins=CORS_ORIGINS, methods=["GET", "POST"])
 
 mpi_comm = None
 
 
+def _check_security():
+    """
+    Checks the security level of the NEST Server instance.
+    """
+
+    msg = []
+    if AUTH_DISABLED:
+        msg.append("AUTH:\tThe authorization is disabled.")
+    if "*" in CORS_ORIGINS:
+        msg.append("CORS:\tAllowed origins is not restricted.")
+    if EXEC_CALL_ENABLED:
+        msg.append("EXEC CALL:\tAny code scripts can be executed!")
+        if RESTRICTION_DISABLED:
+            msg.append("RESTRICTION: Code scripts will be executed without a restricted environment.")
+
+    if len(msg) > 0:
+        print(
+            "WARNING: You chose to disable important access restrictions!\n"                                                    
+            " This allows other computers to execute code on this machine as the current user!\n"                               
+            " Be sure you understand the implications of these settings and take"                                                  
+            " appropriate measures to protect your runtime environment!"                                                        
+        ) 
+        print("\n - ".join([" "] + msg) + "\n")
+
+
+@app.before_request
+def _setup_auth():
+    """
+    Authentication function that generates and validates the NESTServerAuth header with a
+    bearer token.
+
+    Cleans up references to itself and the running `app` from this module, as it may be
+    accessible when the code execution sandbox fails.
+    """
+    try:
+        # Import the modules inside of the auth function, so that if they fail the auth
+        # returns a forbidden error.
+        import inspect  # noqa
+        import gc  # noqa
+        import time  # noqa
+        import hashlib  # noqa
+        import hmac  # noqa
```

Comment:
The PyPi package literally says
> This is a stand alone packaging of the hashlib compatible hmac library included with Python 2.5 so that it can be used together with the hashlib backport on older versions of Python (tested on 2.4). Not needed with Python 2.5 or later.

I guess 2.5 or later applies, so this needs a replacement.